libs/ai-endpoints/langchain_nvidia_ai_endpoints/chat_models.py

```diff
@@ -318,7 +318,7 @@ def _get_generation(
         **kwargs: Any,
     ) -> dict:
         """Call to client generate method with call scope"""
-        kwargs["stop"] = kwargs.get("stop", self.stop)
```

Comment:
why this change?

Reply:
`_generate` has `stop: Optional[List[str]] = None` and always calls `_get_generation(..., stop=stop, ...)`

this means `"stop"` is always in `kwargs`, even if the value is `None`. as a result, `kwags.get("stop", self.stop)` never returns the `self.stop` default.

to get the default of `self.stop` we need to see if `stop` is `not None`

Reply:
Yeah, makes sense

Reply:
we could pass `stop=stop or self.stop` from `_generate`. i originally went with that before settling on this because it is consistent w/ how the stream and async methods work.

we should really mark the stop property as deprecated.